src/ui/backlogmenu.cpp

```diff
@@ -43,109 +50,228 @@ BacklogMenu::BacklogMenu() {
 }
 
 void BacklogMenu::Show() {
-  if (State != Shown) {
+  if (State == Hidden) {
     State = Showing;
     FadeAnimation.StartIn();
     MainItems->Show();
+
     if (!MainItems->Children.empty()) {
       auto el = MainItems->Children.back();
       FocusStart[FDIR_UP] = el;
       FocusStart[FDIR_DOWN] = el;
     }
-    if (UI::FocusedMenu != 0) {
+
+    if (UI::FocusedMenu != nullptr) {
       LastFocusedMenu = UI::FocusedMenu;
       LastFocusedMenu->IsFocused = false;
     }
-    IsFocused = true;
     UI::FocusedMenu = this;
+
+    directionalButtonHeldTime = 0.0f;
+
+    // Set scrollbar back to default position
+    if (ItemsHeight > MainItems->RenderingBounds.Height) {
+      PageY = MainScrollbar->EndValue;
+      MainScrollbar->Update(0);
+      MainItems->MoveTo(glm::vec2(EntriesStart.x, PageY));
+    }
   }
 }
 void BacklogMenu::Hide() {
-  if (State != Hidden) {
+  if (State == Shown) {
     State = Hiding;
     FadeAnimation.StartOut();
-    MainItems->Hide();
-    if (LastFocusedMenu != 0) {
+    directionalButtonHeldTime = 0.0f;
+
+    if (LastFocusedMenu != nullptr) {
       UI::FocusedMenu = LastFocusedMenu;
       LastFocusedMenu->IsFocused = true;
     } else {
-      UI::FocusedMenu = 0;
+      UI::FocusedMenu = nullptr;
     }
-    IsFocused = false;
+  }
+}
+
+inline bool inVerticalHoverBounds(const Widget* entry) {
```

Comment:
Remove the inline and capitalize function name

Reply:
Done